### PR TITLE
Adding an ill-defined profile error for 0-logs

### LINKED ISSUE #1614: create_profile with extrema at zero of a log-scaled field produces NaNs and seg faults
### Bug report

**Bug summary**

This was originally reported by @stonnes to the website repo as https://github.com/yt-project/website/issues/22. I've reposted it here and trimmed down her example a bit to make it easier to see what's going wrong.

Using create_profile with an explicit extrema set to zero of a field that yt thinks should be log-scaled produces nonsense results.

**Code for reproduction**

```python
import yt

ds = yt.load('IsolatedGalaxy/galaxy0030/galaxy0030')
disk = ds.disk([0.5,0.5,0.5],[0,0,-1.0],(40,"kpc"),(2,"kpc"))

prof = yt.create_profile(disk,'radius',('gas','density'), units={'radius':'kpc'}, extrema={'radius':((0.0,'kpc'),(40.0,'kpc'))},weight_field='ones\
')

radius = prof.x
mean = prof['gas','density']
std = prof.standard_deviation['gas','density']
print(radius)
print(mean)
print(std)
```

**Actual outcome**

On my machine, I sometimes get outputs like this:

```
[ nan  nan  nan  nan  nan  nan  nan  nan  nan  nan  nan  nan  nan  nan  nan
  nan  nan  nan  nan  nan  nan  nan  nan  nan  nan  nan  nan  nan  nan  nan
  nan  nan  nan  nan  nan  nan  nan  nan  nan  nan  nan  nan  nan  nan  nan
  nan  nan  nan  nan  nan  nan  nan  nan  nan  nan  nan  nan  nan  nan  nan
  nan  nan  nan  nan] kpc
[ 0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.
  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.
  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.
  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.] g/cm**3
[ 0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.
  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.
  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.
  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.] g/cm**3
```

And sometimes get segfaults with the following python traceback:

```
Fatal Python error: Segmentation fault

Current thread 0x00007fff95d80340 (most recent call first):
  File "/Users/goldbaum/Documents/yt-git-fixes/yt/utilities/parallel_tools/parallel_analysis_interface.py", line 508 in parallel_objects
  File "/Users/goldbaum/Documents/yt-git-fixes/yt/data_objects/profiles.py", line 126 in add_fields
  File "/Users/goldbaum/Documents/yt-git-fixes/yt/data_objects/profiles.py", line 1095 in create_profile
  File "yt_diskprofile.py", line 6 in <module>
[1]    2128 segmentation fault  python -X faulthandler yt_diskprofile.py
```

**Expected outcome**

yt should either be issuing an error or should automatically switch to using a linear scaling when a user explicitly supplies an `extrema` value with one end set to zero.

**Version Information**

  * Operating System: MacOS
  * Python Version: 3.6
  * yt version: 3.5-dev
  * Other Libraries (if applicable):

### PR DESCRIPTION
**Pull request by MatthewTurk**
This fixes #1614.  Rather than intermittently failing, throw an error.

### CODE REVIEW

File: yt/data_objects/profiles.py [hunk 1]
```diff
@@ -1157,6 +1157,8 @@ def create_profile(data_source, bin_fields, fields, n_bins=64,
 
     args = [data_source]
     for f, n, (mi, ma), l in zip(bin_fields, n_bins, ex, logs):
+        if mi == 0 and l:
```
Comment (ngoldbaum) [inline yt/data_objects/profiles.py]: Can you make this `<= 0`?

Comment (MatthewTurk) [reply yt/data_objects/profiles.py]: yup, good catch

**Comment by ngoldbaum** [comment]
@yt-fido test this please

**Comment by ngoldbaum** [comment]
@yt-fido test this please

**Pull request merged**